avocado/core/test.py

```diff
@@ -569,11 +569,7 @@ def _run_avocado(self):
             raise exceptions.TestSkipError(details)
         except exceptions.TestCancel as details:
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
-            skip_illegal_msg = ('Calling cancel() in setUp() '
-                                'is not allowed in avocado, you '
-                                'must fix your test. Original cancel exception: '
-                                '%s' % details)
-            raise exceptions.TestError(skip_illegal_msg)
+            raise
```

Comment:
This means `tearDown` won't be executed, I thought we changed it (in https://trello.com/c/viBJIEwI/941-add-support-to-cancel-test-execution there is this sentence - `teardown() is always run when using self.cancel()`)

anyway as we don't run `tearDown` on `setUp` failure I'm fine with that. (more-like I'd like to change that, I'll try to write an RFC)